### PR TITLE
feat(ui): Implement core UI components

### LINKED ISSUE #34: [Feature] Implement core UI components
## ⭐ Requested Feature / Idea

Implement **core UI components** that serve as the foundation for Framix:
`Spinner`, `TextButton`, `IconButton`, and `Typography`. <br />

Goals: accessible-by-default, token-driven styling (Tailwind CSS v4), consistent API, and clear docs/tests.

<br />

## 💡 Details

**Component Type**

- [x] Basic UI Component (`Button`, `Input`, etc.)
- [ ] Layout Component (`Grid`, `Container`, etc.)
- [ ] Complex Component (`Modal`, `Dropdown`, etc.)
- [ ] Utility/Hook
- [ ] Design Token/Theme Enhancement
- [x] Documentation Improvement

<br />

**Requirements**

- [x] Accessibility compliance ([WCAG](https://w3.org/WAI/standards-guidelines/wcag))
- [x] Mobile responsive
- [x] Dark mode support
- [x] TypeScript definitions
- [x] Storybook stories

<br />

**Scope (Components & API draft)**

- **Spinner**
  - `size`: `"sm" | "md" | "lg"` (default: `"md"`)
  - `color`: semantic tokens (e.g., `"default" | "primary" | "success" | "warning" | "danger"`)
  - `label`: `string | null` (default `"Loading..."`; `null` hides text)
  - `inline`: `boolean` (inline vs block)
  - a11y: `role="status"`, `aria-live="polite"`, label fallback when text hidden

- **TextButton**
  - `variant`: `"solid" | "outline" | "ghost" | "link"`
  - `intent`: `"primary" | "neutral" | "success" | "warning" | "danger"`
  - `size`: `"sm" | "md" | "lg"`
  - `disabled`, `loading`
  - optional `leadingIcon`, `trailingIcon` (ReactNode)
  - a11y: focus-visible ring, disabled semantics, `aria-busy` when loading

- **IconButton**
  - `icon`: `ReactNode` (required)
  - `ariaLabel`: `string` (required for screen readers)
  - `variant`, `intent`, `size` (same as TextButton)
  - `shape`: `"square" | "circle"`
  - a11y: label-only icon button meets WCAG name/role/value

- **Typography**
  - `as`: HTML tag polymorphism (e.g., `"h1" | "h2" | "p" | "span" | "code"`)
  - `variant`: `"h1"…"h6" | "body" | "caption" | "code"`
  - `weight`, `align`, `truncate`, `muted`
  - maps to Tailwind v4 typographic tokens; ensure readable contrast in dark/light

<br />

**Acceptance Criteria**

- Storybook stories with Controls for every prop; MDX usage examples
- Unit tests (Vitest + Testing Library) covering a11y roles/labels, variants, sizes
- Visual regression enabled (Chromatic) for at least Spinner & Buttons
- Token-driven styles (no hard-coded hex); dark mode verified
- JSDoc (one-line summary + minimal examples) for public APIs
- Tree-shakeable exports and zero peer-dep warnings

<br />

**Out of Scope (v0.3.0)**

- Button groups, split buttons, loading skeletons
- Advanced typography (prose, rich text) and motion presets beyond Spinner

<br />

**Use Case**

- Show async progress (data fetch, route transitions) with `Spinner` (with/without label)
- Primary actions in forms and dialogs via `TextButton`
- Icon-only actions (close, menu, like) via `IconButton`
- Consistent headings/body text across pages using `Typography`

<br />

## 🎨 Design Reference

- Figma: Spinner variants (sizes, colors, with/without label) — provided asset<br />
- Figma: Buttons & Typography — to be created before implementation (follow token palette & spacing scale)

<br />

## 💬 Additional Notes (Screenshots, URLs, etc.)

- Release target: **v0.3.0**
- Branch: `feature/core-ui-components`
- Tasks (suggested):
  - [x] Figma spec: Buttons & Typography
  - [x] Implement Spinner + stories/tests
  - [x] Implement TextButton + stories/tests
  - [x] Implement IconButton + stories/tests
  - [x] Implement Typography + stories/tests
  - [x] Chromatic snapshot baseline update
  - [ ] Docs: usage guidelines & a11y notes

### PR DESCRIPTION
**Pull request by herokwon**
## 🔍 Overview

> This pull request introduces a core set of UI components to Framix’s design system. <br />
Added components: `Button`, `IconButton`, `LinkButton`, `Text`, `Heading`, `Spinner`, `Alert`, and `Tooltip`. <br />
Improvements also include enhanced polymorphic type support, utility function updates, color/status/size props, accessibility enhancements, and story/tests coverage.

<br />

## 🛠 Changes

- Added 8 new UI components with prop support for `variant`, `color/status`, `size`, etc.  
- Updated and refined types in `src/types/common` for ElementVariant / ElementStatus.  
- Improved `Button` styling and behavior: semibold font in filled variant, fullWidth prop, etc.  
- Introduced `Tooltip` and `Alert` UI components with stories/tests.  
- Utility improvements (`cn` function, is-local-url, etc.), updated CSS variables / design tokens.  
- Storybook argTypes / controls updated for color, size, spacing.  

<br />

## ❗ Related Issues

- #34

<br />

## 💬 Additional Notes (Screenshots, URLs, etc.)

- ⚠️ No changeset found so version bump may not occur automatically; if intended, add a changeset before merging.
- Check all components for consistent prop names: `variant`, `color` / `status`, `size`, accessibility props (`aria-`, roles).
- Confirm that filled variants use semibold font weight (600) for better visual emphasis.
- Validate that all stories include controls for variant/color/size and that tests cover edge cases.

### CODE REVIEW

File: src/components/ui/Spinner.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    <span
+      {...props}
+      data-testid={`${testId}-wrapper`}
+      className={cn(
+        props.className,
+        'text-foreground-light dark:text-foreground-dark animate-spin-wrap',
+        position !== 'inline' &&
+          `${
+            position === 'global' ? 'fixed' : 'absolute'
+          } top-1/2 left-1/2 z-50 -translate-1/2`,
```
[the remark this answers is not in the document]
Comment (herokwon) [reply src/components/ui/Spinner.tsx:29]: https://tailwindcss.com/docs/translate#examples

File: src/components/ui/Button.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
+import type { Meta, StoryObj } from '@storybook/react-vite';
+
+import { fn } from 'storybook/test';
```
[the remark this answers is not in the document]
Comment (herokwon) [reply src/components/ui/Button.stories.tsx:3]: https://storybook.js.org/docs/writing-tests/interaction-testing#spying-on-functions-with-fn

File: src/components/ui/IconButton.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    icon: Check,
+    variant: 'filled',
+    color: 'default',
+    size: 'md',
+    shape: 'circle',
+    isDisabled: false,
+    isSelected: false,
+    isLoading: false,
+    onClick: fn(),
+  },
```
[the remark this answers is not in the document]
Comment (herokwon) [reply src/components/ui/IconButton.stories.tsx:31]: That's not accessible label. It's `aria-label` for accessibility.
